### PR TITLE
Created skeleton transaction web view

### LINKED ISSUE #1475: As a user I can access the Transactions web view
We should allow a user to access the Transactions view from the extension. In the current designs, this can be done by right-clicking on a smart contract in the gateways view.

image

WIP design for the Transactions view:

image

### PR DESCRIPTION
**Pull request by erin-hughes**
Closes #1475 and contributes to #1155 

Signed-off-by: Erin Hughes <Erin.Hughes@ibm.com>

### CODE REVIEW

File: enzyme/tests/Page.test.tsx [hunk 1]
```diff
@@ -0,0 +1,28 @@
+import React, { Component } from 'react';
```
Comment (Jakeeyturner) [inline enzyme/tests/Page.test.tsx]: As @cazfletch mentioned, we should probably make `Page` and `Sidebar` less generic as they're unlikely to be imported/used by other components.

File: enzyme/tests/Page.test.tsx [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+    });
+
+    it('should render the expected snapshot', async () => {
+        const component: any = renderer
+            .create(<Page activeSmartContract="penguinContract@0.0.1"/>)
+            .toJSON();
+        expect(component).toMatchSnapshot();
+    });   
+
+    afterEach(async () => {
```
Comment (Jakeeyturner) [inline enzyme/tests/Page.test.tsx]: Could we move `afterEach` below `beforeEach` but before `it` as we do it similarly in our extension tests.

File: enzyme/tests/Sidebar.test.tsx [hunk 3]
```diff
@@ -0,0 +1,30 @@
+import React, { Component } from 'react';
+import renderer from 'react-test-renderer';
+import Sidebar from '../../src/components/Sidebar/Sidebar';
+import chai from 'chai';
+import sinon from 'sinon';
+import sinonChai from 'sinon-chai';
+chai.should();
+chai.use(sinonChai);
+const should: Chai.Should = chai.should();
```
Comment (Jakeeyturner) [inline enzyme/tests/Sidebar.test.tsx]: You can remove this line as we do `chai.should()` above.

File: src/App.tsx [hunk 4]
```diff
@@ -34,8 +33,7 @@ class App extends React.Component<any> {
                 <Router>
                     <div>
                     <Route render={() => <Redirect push to={this.state.redirectPath}/>}></Route>
-                    <Route path='/one' component={OneComponent}></Route>
-                    <Route path='/two' component={TwoComponent}></Route>
+                    <Route path='/one' render={() => <Page activeSmartContract="penguinContract@0.0.1"/>}></Route>
```
Comment (Jakeeyturner) [inline src/App.tsx]: Probably worth changing the path `/one` to `/transaction` or something.
If `OneComponent` isn't being used anymore we can probably delete it.

Comment (Jakeeyturner) [reply src/App.tsx]: Same with `TwoComponent`!

File: src/components/Page/Page.scss [hunk 5]
```diff
@@ -0,0 +1,63 @@
+@import '@carbon/layout/scss/_spacing';
+@import 'node_modules/carbon-components/scss/components/accordion/_accordion';
+@import 'node_modules/carbon-components/scss/components/link/link';
+
+body {
```
Comment (Jakeeyturner) [inline src/components/Page/Page.scss]: Is this styling something we should be doing at the App level instead?

File: src/components/Page/Page.scss [hunk 6]
```diff
@@ -0,0 +1,63 @@
...
+
+    h5 {
+        padding-bottom: $spacing-06;
+    }
+}
+
+.page-contents {
+    display: flex;
+    flex-direction: column;
+    margin-left: 325px;
```
Comment (Jakeeyturner) [inline src/components/Page/Page.scss]: For all page layout, we should probably try to use the carbon grid

File: src/components/Page/Page.tsx [hunk 7]
```diff
@@ -0,0 +1,65 @@
...
+                            <h5>Smart Contracts</h5>
+                            <p>To switch to the web view for a different smart contract, select the smart contract below</p>
+                            <Link disabled>{this.state.activeSmartContract}</Link>
+                            <br/>
+                            <Link>congaContract@0.0.1</Link>
+                        </div>
+                        <div className="contents-right" id="contents-right">
+                            <h5>Getting Started</h5>
+                            <Accordion>
+                                <AccordionItem title="Create a new transaction"></AccordionItem>
```
Comment (Jakeeyturner) [inline src/components/Page/Page.tsx]: When changing the theme from dark to light, this text doesn't seem to change colour.

**Comment by erin-hughes** [comment]
Yesterday Caroline asked me to get the build to publish the results of the jest tests like we do for the unit tests. I _think_ I have the test report publishing, but am not sure how to tackle publishing the coverage report. Will have another go on Monday 👍

File: package.json [hunk 8]
```diff
@@ -734,6 +734,14 @@
                     "command": "gatewaysExplorer.testSmartContractEntry",
                     "when": "view == gatewaysExplorer && viewItem == blockchain-instantiated-multi-contract-item || viewItem == blockchain-instantiated-contract-item || viewItem == blockchain-contract-item || viewItem == blockchain-instantiated-unknown-item"
                 },
+                {
+                    "command": "reactPage.open",
```
Comment (Jakeeyturner) [inline package.json]: Might be worth change `reactPage.open` to `transactionPage.open` or something similar.

File: src/components/TransactionViewSidebarPanel/TransactionViewSidebarPanel.tsx [hunk 9]
```diff
@@ -0,0 +1,62 @@
...
+    public importTxn() {
+        console.log('import an existing transaction');
+    }
+
+    private getContents(): any {
+        let panelTSX: any;
+
+        if (this.state.panelType === 'buttons') {
+            panelTSX = (
+                <div className="panel-container" id="panel-container">
```
Comment (Jakeeyturner) [inline src/components/TransactionViewSidebarPanel/TransactionViewSidebarPanel.tsx]: Spotted it in a few places - I don't think you want an id & a class. I think you could probably remove these ids.

File: src/components/TransactionViewPage/TransactionViewPage.tsx [hunk 10]
```diff
@@ -0,0 +1,67 @@
...
+        }
+    }
+
+    public render() {
+        return (
+            <div className="page-container bx--grid" id="page-container">
+                <div className="inner-container bx--row">
+                    <Sidebar/>
+                    <div className="page-contents bx--col" id="page-contents">
+                        <div className="titles-container" id="titles-container">
```
Comment (Jakeeyturner) [inline src/components/TransactionViewPage/TransactionViewPage.tsx]: Not sure you need these ids as well!

**Pull request merged**